File: front/int18/src/app/services/user-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../model/User.model';

@Injectable({
  providedIn: 'root'
})
export class UserService {

private apiUrl = 'http://localhost:8081/api/users'; 

  constructor(private http: HttpClient) { }


  getAllUsers(page: number = 0, size: number = 10): Observable<any> {
    return this.http.get(`${this.apiUrl}?page=${page}&size=${size}`);
  }


  getUserById(idU: number): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/${idU}`);
  }

  
  inscrire(user: User): Observable<boolean> {
    return this.http.post<boolean>(`${this.apiUrl}/inscrire`, user);
  }

  login(username: string, motdepasse: string): Observable<User> {
    return this.http.post<User>(`${this.apiUrl}/login`, { nomUser: username, motdepasse });
  }


  logout(idU: number): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/logout/${idU}`, null);
  }

  createReservation(idU: number, res: any): Observable<boolean> {
    return this.http.post<boolean>(`${this.apiUrl}/${idU}/reservations`, res);
  }


  createUser(user: User): Observable<User> {
    return this.http.post<User>(this.apiUrl, user);
  }


  updateUser(idU: number, user: User): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/${idU}`, user);
  }

  deleteUser(idU: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${idU}`);
  }}

File: lieuRes/lieuRes/src/main/java/com/mss/lieuRes/service/UserService.java
package com.mss.lieuRes.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.mss.lieuRes.entities.Reservation;
import com.mss.lieuRes.entities.User;
public interface UserService {
    Page<User> getAllUsers(int page, int size);
    com.mss.lieuRes.entities.User getUserById(Integer id);
    boolean inscrire(User user);
    com.mss.lieuRes.entities.User login(String nomUser, String motdepasse);
    void logout(Integer idU);
    boolean saveRes(Integer idU, Reservation res);
    boolean updateRes(Integer idU, Reservation res, Integer idR);
    boolean suppRes(Integer idU, Integer idR);
    List<Reservation> getAllRes(Integer idU);
    User createUser(User user);
    User updateUser(Integer id, User user);
    void deleteUser(Integer id);

  
}
